### PR TITLE
[WIP][Doc] Redirect library examples pages to example gallery

### LINKED ISSUE #41616: [docs infra] Replace Examples pages under individual libraries (Data, Serve, Train, Tune, RLlib) with a pre-filtered Example Gallery page
### Description

The Examples pages under each library are not standardized but are curated. They should match a filtered view of the Example Gallery. For example, the Train>Examples page should have the same examples as Example Gallery with the Model training filter on. 

### Link

* https://docs.ray.io/en/latest/ray-core/examples/overview.html
* https://docs.ray.io/en/latest/data/examples/index.html
* https://docs.ray.io/en/latest/train/examples.html
* https://docs.ray.io/en/latest/tune/examples/index.html
* https://docs.ray.io/en/latest/rllib/rllib-examples.html

### PR DESCRIPTION
**Pull request by peytondmurray**
## Why are these changes needed?

This PR redirects users who click on the examples links that are nested under Ray libraries to the example gallery, with the appropriate filters already applied which show only examples related to the appropriate Ray library.

The content of

- `ray-core/examples/overview`
- `data/examples/index`
- `train/examples`
- `tune/examples/index`
- `rllib/rllib-examples`

has been deleted because we're redirecting users to the examples gallery anyway.

## Related issue number

Closes #41616.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a
           method in Tune, I've added it in `doc/source/tune/api/` under the
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -424,6 +424,29 @@ def add_custom_assets(
...
+def set_examples_redirects(
+    app: sphinx.application.Sphinx,
+    pagename: str,
+    templatename: str,
+    context: Dict[str, Any],
+    doctree: nodes.Node,
+):
+    """Insert JS which redirects users to the example gallery for certain pages."""
+    tags = {
+        "ray-core/examples/overview": "",
```
Comment (maxpumperla) [inline doc/source/conf.py:436]: I think we need a Ray Core tag to make this work

Comment (peytondmurray) [reply doc/source/conf.py:436]: @angelinalg What should the tags be for each of these sections?

Comment (angelinalg) [reply doc/source/conf.py:436]: I thought there was talk about a tag for each library. Something like `core`, `data`, `serve`, `train`, `rllib`, `cluster`?

Comment (peytondmurray) [reply doc/source/conf.py:436]: Nobody went through the examples in the example gallery systematically to assign these tags. I started putting them on examples I touched (and was sure about) for other PRs a while back, anticipating that we'd want to do this at some point.

Comment (angelinalg) [reply doc/source/conf.py:436]: I think there are already tags that correspond to the different libraries. Can we just use those? Eventually, we may need a second set: one for the library and one for the workload.

Comment (peytondmurray) [reply doc/source/conf.py]: Fixed.

**Comment by peytondmurray** [comment]
In testing this, I realized that we don't currently have buttons that can filter for these libraries. With the way that we currently have the example gallery page, we can't actually filter on these tags unless we provide buttons for them. Is that something we want to do here @angelinalg?

**Comment by angelinalg** [comment]
Is this something that you and @simran-2797 can explore, to determine the best UX?

**Comment by peytondmurray** [comment]
Sounds good to me - @simran-2797 I'm happy to make additional buttons for these tags. Do we have appropriate icons for the ray library buttons?

**Comment by peytondmurray** [comment]
After a discussion with @simran-2797 this effort is on hold until further design work on the example gallery is complete.

**Comment by simran-2797** [comment]
As @peytondmurray said, I'm doing some more design work this week on the example gallery. Got some feedback from @matthewdeng so trying to include that here.

**Comment by peytondmurray** [comment]
So strictly speaking this doesn't make any changes to the example gallery. @simran-2797 how do you want to proceed? We're still going to want to redirect people to the example gallery from the separate library example pages, right?

**Comment by peytondmurray** [comment]
Closing, because after discussion we will not be taking this approach.

**Pull request closed without merging**